src/binarySearch.js

```diff
@@ -0,0 +1,35 @@
+const binarySearch = (array , number) => {
+  let low = 0,
+      high = (array.length - 1),
+      mid
+
+  while ( array[mid] != number && low <= high ) {
+    mid = Math.floor( (low + high) / 2 )
+    if ( array[mid] < number ) {
+      low = mid + 1
+    } else {
+      high = mid - 1
+    }
+  }
+
+  return array[mid] === number ? mid : 'Not Found Please Try Again!!!'
+}
+
+const recursiveBinarySearch = (array, number, low, high) => {
+  if ( low > high ) {
+    return - 1
+  }
+  let mid = Math.floor(( high + low ) / 2 )
+
+  if ( number === array[mid] ) {
+    return mid
+  } else if ( array[mid] < number ) {
+    low = mid + 1
```

Comment:
Instead of mutating the value passed in, I would just perform this calculation inline:
```js
return recursiveBinarySearch( array, number, mid + 1, high )
// or
return recursiveBinarySearch( array, number, low, mid - 1 )
```